src/translate/normalize.py

```diff
@@ -366,10 +379,13 @@ def build_exploration_rules(task):
         proxy.build_rules(result)
     return result
 
-def condition_to_rule_body(parameters, condition):
+def condition_to_rule_body(parameters, condition, numeric_expression=None):
```

Comment:
I assume the numeric expression here is still always primitive? Maybe also rename the parameter to the more explicit `primitive_numeric_expression`. Either way I would name it the same as the variable in the calling code.

Reply:
This is what I referred to in my other comment. Variables are now named the same and I added typing information to make things clearer.